tests/test_batch.py

```diff
@@ -72,7 +72,11 @@ def test_create_batch_and_wait(self, request_mock, wait, fetch_results):
         )  # the batch_id used in the mock data
         assert batch.sequence_builder == self.pulser_sequence
         assert batch.complete
-        assert batch.jobs
+        assert batch.ordered_jobs
+        assert batch.ordered_jobs[0].id == self.job_id
+        assert batch.ordered_jobs[0].result == self.job_result
+        assert batch.ordered_jobs[0].full_result == self.job_full_result
+        # attribute 'jobs' to be removed
```

Comment:
Do you want to test the warning here ?

Is the warning message going to appear at each iteration step?That might be quite annoying for the user..

Reply:
Yes I also forgot to add it by it was planned

Reply:
For the user, I have put a stacklevel of 1 which shouldn't spam them that much